### PR TITLE
feat(dws): add new resource to copy automated snapshot

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
1. Extract query and delete methods as common method.
2. Add new resource to copy automated snapshot.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. extract query and delete methods as common method.
2. add new resource to copy automated snapshot.
3. add related to document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```

 make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccSnapshotCopy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccSnapshotCopy_basic -timeout 360m -parallel 4
=== RUN   TestAccSnapshotCopy_basic
=== PAUSE TestAccSnapshotCopy_basic
=== CONT  TestAccSnapshotCopy_basic
--- PASS: TestAccSnapshotCopy_basic (23.84s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       23.934s

make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDwsSnapshot_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDwsSnapshot_basic -timeout 360m -parallel 4
=== RUN   TestAccDwsSnapshot_basic
=== PAUSE TestAccDwsSnapshot_basic
=== CONT  TestAccDwsSnapshot_basic
--- PASS: TestAccDwsSnapshot_basic (336.99s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       337.091s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    The **huaweicloud_dws_snapshot** resource does not exist.
   image
   The **huaweicloud_dws_snapshot_copy** resource does not exist.
    image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found

    The **huaweicloud_dws_snapshot** resource does not exist.
    image
    The **huaweicloud_dws_snapshot_copy** resource does not exist.
    image

### CODE REVIEW

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_copy_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/dws"
+)
+
+func getCopiedSnapshotFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	client, err := cfg.NewServiceClient("dws", region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DWS Client: %s", err)
```
Comment (ruwenqiang123) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_copy_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating DWS client: %s", err)
```

Comment (ruwenqiang123) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_copy_test.go]: other place ditto

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDwsSnapshot_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDwsSnapshot_basic -timeout 360m -parallel 4
=== RUN   TestAccDwsSnapshot_basic
=== PAUSE TestAccDwsSnapshot_basic
=== CONT  TestAccDwsSnapshot_basic
--- PASS: TestAccDwsSnapshot_basic (328.10s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       328.166s


make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccSnapshotCopy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccSnapshotCopy_basic -timeout 360m -parallel 4
=== RUN   TestAccSnapshotCopy_basic
=== PAUSE TestAccSnapshotCopy_basic
=== CONT  TestAccSnapshotCopy_basic
--- PASS: TestAccSnapshotCopy_basic (23.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       23.405s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDwsSnapshot_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDwsSnapshot_basic -timeout 360m -parallel 4
=== RUN   TestAccDwsSnapshot_basic
=== PAUSE TestAccDwsSnapshot_basic
=== CONT  TestAccDwsSnapshot_basic
--- PASS: TestAccDwsSnapshot_basic (292.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       292.438s


make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccSnapshotCopy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccSnapshotCopy_basic -timeout 360m -parallel 4
=== RUN   TestAccSnapshotCopy_basic
=== PAUSE TestAccSnapshotCopy_basic
=== CONT  TestAccSnapshotCopy_basic
--- PASS: TestAccSnapshotCopy_basic (21.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       22.073s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDwsSnapshot_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDwsSnapshot_basic -timeout 360m -parallel 4
=== RUN   TestAccDwsSnapshot_basic
=== PAUSE TestAccDwsSnapshot_basic
=== CONT  TestAccDwsSnapshot_basic
--- PASS: TestAccDwsSnapshot_basic (292.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       292.438s

make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccSnapshotCopy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccSnapshotCopy_basic -timeout 360m -parallel 4
=== RUN   TestAccSnapshotCopy_basic
=== PAUSE TestAccSnapshotCopy_basic
=== CONT  TestAccSnapshotCopy_basic
--- PASS: TestAccSnapshotCopy_basic (21.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       22.073s

File: huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go [hunk 2]
```diff
@@ -268,50 +264,35 @@ func resourceDwsSnapshotRead(_ context.Context, d *schema.ResourceData, meta int
...
-		errCode, err := jmespath.Search(`error_code`, apiErr)
-		if err != nil {
-			return fmt.Errorf("error parse errorCode from response body: %s", err.Error())
-		}
-
-		if errCode == `DWS.5149` {
-			return golangsdk.ErrDefault404{}
-		}
+// GetSnapshotById is a method used to delete snapshot.
+func DeleteSnapshotById(client *golangsdk.ServiceClient, snapshotId string) error {
```
Comment (Lance52259) [inline huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go]:
```suggestion
func deleteSnapshotById(client *golangsdk.ServiceClient, snapshotId string) error {
```

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccSnapshotCopy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccSnapshotCopy_basic -timeout 360m -parallel 4
=== RUN   TestAccSnapshotCopy_basic
=== PAUSE TestAccSnapshotCopy_basic
=== CONT  TestAccSnapshotCopy_basic
--- PASS: TestAccSnapshotCopy_basic (27.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       27.678s


make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDwsSnapshot_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDwsSnapshot_basic -timeout 360m -parallel 4
=== RUN   TestAccDwsSnapshot_basic
=== PAUSE TestAccDwsSnapshot_basic
=== CONT  TestAccDwsSnapshot_basic
--- PASS: TestAccDwsSnapshot_basic (325.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       325.374s

File: huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go]: fixed

**Pull request merged**